flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -568,8 +568,8 @@ public void setInitialState(ChainedStateHandle<StreamStateHandle> chainedState,
 	private void restoreState() throws Exception {
 		final StreamOperator<?>[] allOperators = operatorChain.getAllOperators();
 
-		try {
-			if (lazyRestoreChainedOperatorState != null) {
+		if (lazyRestoreChainedOperatorState != null) {
```

Comment:
this `if (lazyRestoreChainedOperatorState != null)` check was moved out of the `try` block because the same check is required in the `finally` block